website/shared/markdown/time.go

```diff
@@ -0,0 +1,129 @@
+package markdown
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/yuin/goldmark/ast"
+	"github.com/yuin/goldmark/parser"
+	"github.com/yuin/goldmark/renderer"
+	"github.com/yuin/goldmark/text"
+	goldutil "github.com/yuin/goldmark/util"
+)
+
+type TimeParser struct {
+	debugEnabled bool
+}
+
+func (p *TimeParser) debug(s string, args ...any) {
+	if p.debugEnabled {
+		log.Printf(s, args...)
+	}
+}
+
+var _ parser.InlineParser = (*TimeParser)(nil)
+
+const (
+	timeStart = '{'
+	timeEnd   = '}'
+	durStart  = '|'
+)
+
+func (p *TimeParser) Trigger() []byte {
+	return []byte{timeStart}
+}
+
+func (p *TimeParser) Parse(parent ast.Node, reader text.Reader, pc parser.Context) ast.Node {
+	if reader.Peek() != timeStart {
+		return nil
+	}
+
+	// eat our beginning
+	reader.Advance(1)
+	_, seg := reader.Position()
+	var i int
+	var indur bool
+	timeRs := []rune{}
+	durRs := []rune{}
+	for {
+		r, _, err := reader.ReadRune()
+		if err != nil {
+			return nil
+		}
+		if r == timeEnd {
+			break
+		} else if r == durStart { // allow a duration, don't include it in output text
+			indur = true
+			continue
+		} else if r == '\n' { // failsafe
+			p.debug("newline encountered")
+			return nil
+		}
+		if indur {
+			durRs = append(durRs, r)
+			continue
+		}
+		timeRs = append(timeRs, r)
+		i++
+	}
```

Comment:
This lexer seems overly complex. Can't you just search for the '}' and then grab the string between it? Or use a regex...?

You also need to ensure that this doesn't needlessly break if there just happens to be a { in the text somewhere.